### ISSUE TITLE
Formie has broken CraftCMS garbage collection

### ISSUE THREAD

**Issue by cloudgrayau** [verbb/formie#570]
Having Formie enabled breaks CraftCMS garbage collection as follows:

`$ ./craft gc --delete-all-trashed=1
Running garbage collection ... PHP Notice 'yii\base\ErrorException' with message 'Trying to get property 'columns' of non-object'

in /cloudgray/vendor/craftcms/cms/src/services/Fields.php:1044

Stack trace:
#0 /cloudgray/vendor/craftcms/cms/src/services/Fields.php(1044): yii\console\ErrorHandler->handleError()
#1 /cloudgray/vendor/craftcms/cms/src/services/Fields.php(990): craft\services\Fields->_dropOldFieldColumns()
#2 /cloudgray/vendor/craftcms/cms/src/services/Fields.php(938): craft\services\Fields->applyFieldDelete()
#3 /cloudgray/vendor/verbb/formie/src/services/Fields.php(465): craft\services\Fields->deleteField()
#4 /cloudgray/vendor/verbb/formie/src/Formie.php(283): verbb\formie\services\Fields->deleteOrphanedFields()
#5 /cloudgray/vendor/yiisoft/yii2/base/Event.php(312): verbb\formie\Formie->verbb\formie\{closure:/cloudgray/vendor/verbb/formie/src/Formie.php:281-299}()
#6 /cloudgray/vendor/yiisoft/yii2/base/Event.php(312): ::call_user_func:{/cloudgray/vendor/yiisoft/yii2/base/Event.php:312}()
#7 /cloudgray/vendor/yiisoft/yii2/base/Component.php(637): yii\base\Event::trigger()
#8 /cloudgray/vendor/craftcms/cms/src/services/Gc.php(103): craft\services\Gc->trigger()
#9 /cloudgray/vendor/craftcms/cms/src/console/controllers/GcController.php(80): craft\services\Gc->run()
#10 /cloudgray/vendor/yiisoft/yii2/base/InlineAction.php(57): craft\console\controllers\GcController->actionRun()
#11 /cloudgray/vendor/yiisoft/yii2/base/InlineAction.php(57): ::call_user_func_array:{/cloudgray/vendor/yiisoft/yii2/base/InlineAction.php:57}()
#12 /cloudgray/vendor/yiisoft/yii2/base/Controller.php(181): yii\base\InlineAction->runWithParams()
#13 /cloudgray/vendor/yiisoft/yii2/console/Controller.php(184): craft\console\controllers\GcController->runAction()
#14 /cloudgray/vendor/craftcms/cms/src/console/Controller.php(205): craft\console\controllers\GcController->runAction()
#15 /cloudgray/vendor/yiisoft/yii2/base/Module.php(534): craft\console\controllers\GcController->runAction()
#16 /cloudgray/vendor/yiisoft/yii2/console/Application.php(181): craft\console\Application->runAction()
#17 /cloudgray/vendor/craftcms/cms/src/console/Application.php(89): craft\console\Application->runAction()
#18 /cloudgray/vendor/yiisoft/yii2/console/Application.php(148): craft\console\Application->runAction()
#19 /cloudgray/vendor/yiisoft/yii2/base/Application.php(392): craft\console\Application->handleRequest()
#20 /cloudgray/craft(27): craft\console\Application->run()
#21 {main}
`

I hope that helps.

Cheers.

**Comment by cloudgrayau** [comment]
**Additional info**

Plugin version: 1.4.15
Craft version: 3.7.11

**Comment by engram-design** [comment]
Fixed for the next release. To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.4.15",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [1.4.16](https://github.com/verbb/formie/releases/tag/1.4.16)
